concordium-node/src/bin/cli.rs

```diff
@@ -142,21 +145,42 @@ async fn main() -> anyhow::Result<()> {
 
     // Out-of-band catch-up
     if let Some(ref import_blocks_from) = conf.cli.baker.import_blocks_from {
-        let from: anyhow::Result<String> =
-            download_catchup_files(import_blocks_from, data_dir_path).await;
+        info!("Starting out of band catch-up");
+        if let Err(e) = consensus.import_blocks(import_blocks_from) {
+            if import_stopped.load(atomic::Ordering::Acquire) {
+                info!("Out of band catchup stopped.");
+            } else {
+                info!(
+                    "Could not complete out of band catch-up from {} due to: {:#}.",
+                    import_blocks_from.display(),
+                    e
+                );
+            }
+        } else {
+            info!("Completed out of band catch-up from {}.", import_blocks_from.display());
+        }
+    } else if let Some(download_url) = conf.cli.baker.download_blocks_from.as_ref().cloned() {
+        let from = download_catchup_files(download_url.clone(), data_dir_path).await;
         match from {
-            Ok(from) => {
+            Ok(path) => {
                 info!("Starting out of band catch-up");
-                consensus.import_blocks(from.as_bytes());
-                info!("Completed out of band catch-up");
-                if from != *import_blocks_from {
-                    if let Err(e) = std::fs::remove_file(from) {
-                        error!("Cleaning up downloaded out of band catch-up files failed: {}", e);
-                    };
-                };
+                if let Err(e) = consensus.import_blocks(&path) {
+                    if import_stopped.load(atomic::Ordering::Acquire) {
+                        info!("Out of band catchup stopped.");
+                    } else {
+                        info!("Could not complete out of band catch-up due to: {:#}.", e);
+                    }
+                } else {
+                    info!("Completed out of band catch-up from {}.", download_url);
+                }
+                // attempt to properly clean up the downloaded file.
+                if let Err(e) = path.close() {
+                    error!("Could not delete the downloaded file: {}", e);
```

Comment:
I think this should be factored out to avoid having an overly long `main`. Possibly this should also move to consensus.rs.

Reply:
I do agree about factoring out. Not about consensus. The logic here is about getting the file as it is now. The actual import is already in consensus.

I think this should generally be done differently anyhow, and we should not be downloading and storing the file. We should just stream it if given a URL. But I don't have time to do that now.